### PR TITLE
Bumpup nixpkgs

### PR DESCRIPTION
**Pull request by GTrunSec**
@guaraqe  Is it a common format way to use nixpkgs-fmt here?
1. nix-ci-test
 - nix-build tests -A build -A kernel-tests.core

```
wrapping `/nix/store/8bnbhf8qlqmmb5aswxx0kk9n73c07agq-python3.8-jupyterlab-2.2.9/bin/jupyter-labhub'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
/nix/store/8bnbhf8qlqmmb5aswxx0kk9n73c07agq-python3.8-jupyterlab-2.2.9
/nix/store/pbffkk1ivlvgmiaw32nx6x09g9nshzf2-test-kernel-paths-core
```

2. nix-shell test
 - nix-shell

image

### CODE REVIEW

**Comment by GTrunSec** [comment]
tracking issue on macos #144. other same issue mentioned [here](https://github.com/jupyterlab/jupyterlab/issues/9863)

**Comment by guaraqe** [comment]
@GTrunSec Thanks for trying! Can you explain what you are trying to do with Mac here? Normally, I think these large patches would be better placed in Nixpkgs, why do you think it fits here?

**Comment by GTrunSec** [comment]
So, I did a patch here instead of a modified file. The motivation of to disable the check trash on darwin.

Note: (I only tested the python kernel on darwin)

image

File: nix/disable-check-trash-on-darwin.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+             if sys.platform in {'win32', 'darwin'}:
+-                return True
++                return False
+ 
+             # It's a bit more nuanced than this, but until we can better
+             # distinguish errors from send2trash, assume that we can only trash
+@@ -544,7 +543,7 @@ def is_non_empty_dir(os_path):
+                 # raises let us distinguish permission errors from other errors in
+                 # code. So for now, just let them all get logged as server errors.
+                 send2trash(os_path)
```
Comment (jskrzypek) [inline nix/disable-check-trash-on-darwin.patch:25]: You probably should remove this line as well.

**Comment by jskrzypek** [comment]
This is based on a rather dirty hack from a PR on jupyter/notebook: https://github.com/jupyter/notebook/pull/6016

I don't know that it's the right solution as I think the underlying problem is actually this: https://github.com/NixOS/nixpkgs/issues/105038

**Comment by guaraqe** [comment]
I updated to a newer version on master which fixes the Mac CI.

**Pull request closed without merging**